context/app/static/js/shared-styles/charts/hooks.ts

```diff
@@ -154,6 +155,21 @@ function useOrdinalScale(domain: string[], config: Omit<OrdinalScaleConfig<strin
   return useMemo(() => scaleOrdinal<string, string>({ ...config, domain }), [domain, config]);
 }
 
+const useAriaLabelText = (tooltipData: TooltipData<unknown> | undefined) => {
+  return useMemo(() => {
+    if (!tooltipData || !tooltipHasBarData(tooltipData)) {
+      return `Bar represents`;
+    }
+    const extractedValues = Object.entries(tooltipData.bar.data)
+      .filter(([, value]) => typeof value === 'string')
+      .map(([key, value]) => [key, value as string]);
+    const extractedValuesString = extractedValues.join('  ');
+    const count = tooltipData.bar.data[tooltipData.key] ?? '';
+
+    return `Bar represents ${extractedValuesString} with count ${String(count)}`;
+  }, [tooltipData]);
+};
```

Comment:
This is a good iteration over the first approach, but there's still a few things that stand out to me:

1. Since this is based on the currently hovered tooltip data:
  a. If there is no hovered bar, every bar shows up with the `Bar represents` text and nothing else. This is also encountered when hovering tick labels.
![image](https://github.com/user-attachments/assets/48af2518-9ac8-4237-82c7-d96a1134c740)
  b. When a bar is hovered, every bar gets the same aria label: 
![image](https://github.com/user-attachments/assets/256c8728-936c-4791-9b5d-45657b4f327a)

2. The fallthrough to an incomplete label (`"Bar represents"`) is problematic since it makes it sound like the screenreader stopped reading halfway through; this may frustrate users even more than a blank label.

3. The approach of iterating over all bar data may lead to irrelevant or unformatted fields being pulled in to the label if they are not removed from the input data. Additionally, even fields which are relevant might still be read wrong - e.g. `Bar represents datasetType,Slide-seq with count 46` might not be read correctly due to the `datasetType` word.

---

I would suggest trying an alternative approach: each stacked bar chart could be provided with a function prop which takes in the same generic `Datum` being used as the type constraint for the chart overall and returns a string. This would allow us to define well-formatted labels which will be guaranteed to be relevant to the situation at hand, e.g. adding something like this to the end of `HorizontalStackedBarChartProps` and `VerticalStackedBarChartProps`, similar to how we provide `getBarHref`:

```
  getAriaLabel?: (d: TooltipData<Datum>) => string;
``` 

As an example of how we'd be able to apply this - for the homepage bar chart, we would extend the `ColorOption` array to have unique `getAriaLabel`s for each selected aggregation, e.g.

```
    {
      key: 'assay_type',
      data: assayBuckets,
      dropdownLabel: 'Assay Type',
      keys: getKeysFromAggregatedData(assayBuckets),
      getBarHref: (d) => {
        const organTerms = [String(d.bar.data.organ)];
        return getSearchURL({
          entityType: 'Dataset',
          organTerms,
          assay: d.key,
          assayTypeMap: datasetTypeMap,
        });
      },
      getAriaLabel: (d) => {
        const organ = String(d.bar.data.organ);
        const assay = d.key;
        const count = d.bar.data[d.key];
        return `${count} ${organ} datasets with ${assay} dataset type.`
      }
    },
```

This approach ensures that the same datum is used to generate both the href for the specific bar and the aria label.

Reply:
Thanks for reviewing. These are some great suggestions, will incorporate them.
I was also not comfortable with the fallthrough text, but thought that the the screenReader would only read the label when the link is hovered/clicked.